html/js/product-multilingual.js

```diff
@@ -437,22 +450,32 @@ function update_display(imagefield, first_display) {
 				// alert(event.data.imagefield);
 				$('div[id="ocrbutton_loading_' + imagefield +'"]').html('<img src="/images/misc/loading2.gif" /> ' + lang()["product_js_extracting_" + imagetype]).show();
 				$('div[id="ocrbuttondiv_' + imagefield +'"]').hide();
-				$.post('/cgi/' + imagetype + '.pl',
-						{code: code, id: imagefield, process_image:1, ocr_engine:"google_cloud_vision" }, function(data) {
-
-					$('div[id="ocrbuttondiv_' + imagefield +'"]').show();
-					if (data.status === 0) {
-						$('div[id="ocrbutton_loading_' + imagefield +'"]').html(lang()["product_js_extracted_" + imagetype + "_ok"]);
-						var text_id = imagefield.replace(imagetype, imagetype + "_text");
-						$("#" + text_id).val(data[imagetype + "_text_from_image"]);
-					}
-					else {
-						$('div[id="ocrbutton_loading_' + imagefield +'"]').html(lang()["product_js_extracted_" + imagetype + "_nok"]);
-					}
-					$(document).foundation('equalizer', 'reflow');
-				}, 'json');
-
-				$(document).foundation('equalizer', 'reflow');
+				$.post(
+					'/cgi/' + imagetype + '.pl',
+					{code: code, id: imagefield, process_image:1, ocr_engine:"google_cloud_vision" },
+					null,
+					'json'
+				)
+					.done(function(data) {
+						$('div[id="ocrbuttondiv_' + imagefield +'"]').show();
+						if (data.status === 0) {
+							$('div[id="ocrbutton_loading_' + imagefield +'"]').html(lang()["product_js_extracted_" + imagetype + "_ok"]);
+							var text_id = imagefield.replace(imagetype, imagetype + "_text");
+							$("#" + text_id).val(data[imagetype + "_text_from_image"]);
+						}
+						else {
+							$('div[id="ocrbutton_loading_' + imagefield +'"]').html(lang()["product_js_extracted_" + imagetype + "_nok"]);
+						}
+					})
+					.fail(function() {
+						$('div[id="ocrbuttondiv_' + imagefield +'"]').show();
+						$('div[id="ocrbutton_loading_' + imagefield +'"]').html(lang().job_status_failed);
+					})
+					.always(function() {
+						$(document).foundation('equalizer', 'reflow');
+					});
+
+				//$(document).foundation('equalizer', 'reflow');
```

Comment:
If this is actually not required any more, just remove the code. (Same in the function below.)

Reply:
I didn't want to remove it without being sure it wasn't specified twice for a reason.

Reply:
It doesn't happen at the same time. In practice the reflow is needed when we change the content in a way that may change the overall height of the content. Otherwise the bottom of the content might be hidden by the footer. In this specific case, we are just showing an icon and hiding a button, so I guess we could remove it (the commented out one).

Reply:
also before, the inner one would only have run on success, whereas now it should always run.